tripy/docs/conf.py

```diff
@@ -149,12 +150,7 @@ def process_docstring(app, what, name, obj, options, lines):
                 elif param.kind == inspect.Parameter.VAR_POSITIONAL:
                     pname = "*" + pname
 
-                if pname == "self":
-                    # Don't want a type annotation for the self parameter.
-                    assert (
-                        param.annotation == signature.empty
-                    ), f"Avoid using type annotations for the `self` parameter since this will corrupt the rendered documentation!"
```

Comment:
Could you check if there is actually a negative impact to the documentation from having annotations for `self` parameters?

Reply:
It seems like sphinx deals with it pretty well. The only thing that needs to be checked is there is documentation for self if it is being used. I will add a check for that and add documentation for each case where I add self to the signature.